### PR TITLE
Feature/kp view employees component

### PR DESCRIPTION
**Pull request by kennypanyavong**

### CODE REVIEW

File: src/components/employees/EmployeeDetails.jsx [hunk 1]
```diff
@@ -1,3 +1,85 @@
...
+  // state to hold fetch
+  const [employee, setEmployee] = useState(null)
+  // error if fetch fails
+  const [error, setError] = useState(null)
+
+  useEffect(() => {
+    const getEmployeeDetails = async () => {
+      try {
+        const response = await fetch(
+          `http://localhost:8088/employees/${employeeId}`
```
Comment (herringvoices) [inline src/components/employees/EmployeeDetails.jsx]: Write a function for this in employee services and call it here.

Comment (herringvoices) [reply src/components/employees/EmployeeDetails.jsx]:
```
export const getEmployees = async (employeeId) => {
  try {
const url = employeeId
    ? `[http://localhost:8088/employees/${employeeId}](http://localhost:8088/employees/$%7BemployeeId%7D)`
    : "http://localhost:8088/employees";
    const response = await fetch(url);
    const employees = await response.json();
    return employees;
  } catch (error) {
    console.error("Error fetching employees:", error);
  }
};
```

File: src/components/employees/EmployeeDetails.jsx [hunk 2]
```diff
@@ -1,3 +1,85 @@
...
+        setEmployee(details)
+      } catch (error) {
+        setError(error.message)
+      }
+    }
+
+    getEmployeeDetails()
+  }, [employeeId]) //dependency array to trigger when employeeId changes
+
+  // handle delete employee
```
Comment (herringvoices) [inline src/components/employees/EmployeeDetails.jsx:26]: There is a service for this in employeeServices that you can call inside your handler function.

File: src/components/employees/EmployeeDetails.jsx [hunk 3]
```diff
@@ -1,3 +1,85 @@
+import { useEffect, useState } from "react"
+import { useNavigate, useParams } from "react-router-dom"
+
 export const EmployeeDetails = () => {
-  return <>EmployeeDetails</>;
-};
+  // get employeeId from the url
```
Comment (herringvoices) [inline src/components/employees/EmployeeDetails.jsx:6]: useNavigate() could be set up here:
`  const navigate = useNavigate();`

File: src/components/employees/EmployeeDetails.jsx [hunk 4]
```diff
@@ -1,3 +1,85 @@
...
+  //       `http://localhost:8088/employees/${employeeId}`,
+  //       {
+  //         method: "DELETE",
+  //       }
+  //     )
+  //     if (!response.ok) {
+  //       throw new Error("Failed to delete employee")
+  //     }
+
+  //     useNavigate("/employees/view") //insert route
```
Comment (herringvoices) [inline src/components/employees/EmployeeDetails.jsx]: then here, you would just be using navigate:
`navigate(`/employees/edit/employeeId`)`

File: src/components/employees/EmployeeDetails.jsx (hunk 2 above)
Comment (herringvoices) [reply src/components/employees/EmployeeDetails.jsx:26]: then handleEditEmployee will use call the `updateEmployee()` function passing in the employee object.

**Review by herringvoices — CHANGES_REQUESTED** [review]

**Review by herringvoices — APPROVED** [review]

**Review by joshuaKsmith — APPROVED** [review]

**Pull request merged**